packages/patterns/src/patterns/patternMatchers.js

```diff
@@ -507,15 +519,20 @@ const makePatternKit = () => {
         }
         const { payload: pattPayload } = patt;
         const { payload: specimenPayload } = specimen;
-        const pattKeySet = copyMapKeySet(pattPayload);
-        const specimenKeySet = copyMapKeySet(specimenPayload);
+        const pattKeySet = copyMapKeySet(patt);
+        const specimenKeySet = copyMapKeySet(specimen);
```

Comment:
To aid in understanding, can you briefly describe the problem here, or make an issue for it?

Reply:
The argument of `copyMapKeySet` is supposed to be a CopyMap. At this location, both `patt` and `specimen` are known to be CopyMaps. The payload of a CopyMap is a pair of a keys array and a corresponding values array. Such a pair is not itself a valid CopyMap.

This fix does do redundant work, in that `copyMapKeySet` reverifies that its argument is a CopyMap, and after that only pays attention to the payload anyway. But with the memoes we already have, that should only be a tiny bit of redundant work. Avoid it would make abstraction boundaries a bit messier.

Perhaps one day our static type checking would be precise enough to have flagged this error ;) .

Reply:
Resolving. But let me know if you think it is worth opening a closing an issue for this instead.

Reply:
> opening and closing an issue

No need.  Thanks for the explanation!